src/ScillaContractDeployer.ts

```diff
@@ -76,6 +78,24 @@ function read(f: string) {
   return t;
 }
 
+/// Allows you to change setup parameters. Available params:
+/// gasPrice, gasLimit, attempts, timeout.
+export function updateSetup(args: any) {
+  if (setup === null) {
+    throw new HardhatPluginError("hardhat-scilla-plugin", "Please call the initZilliqa function.");
+  }
+  let newSetup : Setup = {
```

Comment:
`let merged = {...obj1, ...obj2};`

merged is now the union of obj1 and obj2. Properties in obj2 will overwrite those in obj1.

this syntax is nicer and easier to read.

Reply:
I've changed it, but it's not much nicer. If you try the direct approach .. 
let obj : any = { fish, soup }
let result = { .. setup, .. obj }

and soup is undefined, you get setup.soup overridden to undefined :-(